do not start traffic if given net or ifname not found

pperiyasamy opened the pull request:
tgc by default assigns primary interface when given secondary network or interface not found, this commit makes those streams to be counted as not started.

Signed-off-by: Periyasamy Palanisamy <periyasamy.palanisamy@est.tech>

pperiyasamy committed:
do not start traffic if given net or ifname not found

tgc by default assigns primary interface when given secondary network or interface is not found, this commit makes those streams to be counted as not started.

Signed-off-by: Periyasamy Palanisamy <periyasamy.palanisamy@est.tech>

pperiyasamy commented:
/cc @JanScheurich

pperiyasamy committed:
add ip found check for net name

Signed-off-by: Periyasamy Palanisamy <periyasamy.palanisamy@est.tech>

JanScheurich commented on pkg/tgc/tgc.go:
```diff
@@ -415,18 +417,24 @@ func getAvailableNetBatPairings(namespace, podName, pairingStr string) ([]util.B
 			}
-		}
-		if source.IP == "" {
+		} else if source.IP == "" {
```
Actually, the only legal case if neither net nor interface is present in source is that the source interface is the primary eth0 interface. Checking for source.IP=="" is redundant as the IP is never included in the sourceStr. It is only populated by TGC.

pperiyasamy replied:
@JanScheurich, yes, good find! removed` source.IP==""` check now.

pperiyasamy committed:
remove source ip check for primary network case

this also makes to proceed with starting tgen when eth0 not found
so that stream to be counted as not started.

Signed-off-by: Periyasamy Palanisamy <periyasamy.palanisamy@est.tech>

JanScheurich reviewed (approved):
/lgtm
